quisp/modules/QRSA/HardwareMonitor/HardwareMonitor.cc

```diff
@@ -577,123 +577,124 @@ Matrix4cd HardwareMonitor::reconstruct_Density_Matrix(int qnic_id) {
   density_matrix_Y2.real() ).trace(); EV<<"Yerr = "<<Yerr_rate2<<"\n";*/
 }
 
-Matrix4cd HardwareMonitor::extended_reconstruct_Density_Matrix(int qnic_id, int partner) {
+Matrix4cd HardwareMonitor::extended_reconstruct_density_matrix(int qnic_id, int partner) {
   // II
+  auto tommography_data_with_partner = extended_tomography_data[qnic_id][partner];
   double S00 = 1.0;
-  double S01 = (double)extended_tomography_data[qnic_id][partner]["XX"].plus_plus / (double)extended_tomography_data[qnic_id][partner]["XX"].total_count -
-               (double)extended_tomography_data[qnic_id][partner]["XX"].plus_minus / (double)extended_tomography_data[qnic_id][partner]["XX"].total_count +
-               (double)extended_tomography_data[qnic_id][partner]["XX"].minus_plus / (double)extended_tomography_data[qnic_id][partner]["XX"].total_count -
-               (double)extended_tomography_data[qnic_id][partner]["XX"].minus_minus / (double)extended_tomography_data[qnic_id][partner]["XX"].total_count;
+  double S01 = (double)tommography_data_with_partner["XX"].plus_plus / (double)tommography_data_with_partner["XX"].total_count -
```

Comment:
I want you to make it a shorter name. if there's a `tomography_data` variable or similar name, `tomography_data_with_partner` is a good name. 
but now there's no similar variable in this method. so it's ok to `data` or `tomography_data`.
everyone can understand it easily.

Reply:
OK, let me make it shorter.